utils/transformedPayload.js

```diff
@@ -0,0 +1,24 @@
+const transformPayload = (payload, referenceData) => {
```

Comment:
This function mutates the original payload in `req`. If the original payload is used elsewhere mutating it can cause serious issues. Can this be made a pure function ?